compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -821,7 +821,9 @@ struct AnalyzeAttrState<'a> {
 #[inline]
 fn analyze_attr(attr: &hir::Attribute, state: &mut AnalyzeAttrState<'_>) -> bool {
     let mut should_encode = false;
-    if !rustc_feature::encode_cross_crate(attr.name_or_empty()) {
+    if let hir::Attribute::Parsed(p) = attr {
+        should_encode = p.encode_cross_crate();
+    } else if !rustc_feature::encode_cross_crate(attr.name_or_empty()) {
```

Comment:
@jdonszelmann what do you think about this? There's probably more opportunity for simplifying things, since `name_or_empty` is always empty for parsed attrs, but that should be its own PR